### PR TITLE
Switch the way the package compiles assets

### PR DESCRIPTION
**Pull request by luislard**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Switches the way the assets are built for a release

**What is the current behavior?** (You can also link to an open issue here)
An additional package is required in order to get the assets for this package.

**What is the new behavior (if this is a feature change)?**
No additional package is needed in order to get the assets for this package.

## How to create a release

To create a release go to the `<target>-built` branch and create the tag and the release. 

**Example 1: A change is going to be added to `master` branch.**

A developer makes a PR to `master` branch adding a feature.
Once it gets merged a build & push workflow will be triggered.
[bot] This workflow will create a build and push it to `master-built` branch.
[human] Then create a tag on `master-built` and then a release.

**Example 2: A change is going to be added to `1.x` branch.**
A developer makes a PR to `1.x` branch adding a feature.
Once it gets merged a build & push workflow will be triggered.
[bot] This workflow will create a build and push it to `1.x-built` branch.
[human] Then create a tag on `1.x-built` and then a release.

## Branch protection
We will protect `-built` branches to ensure we don't accidentally delete them

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

**Other information**:

### CODE REVIEW

File: .github/workflows/build-to-release-branches.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Build to release branches
+
+on:
+  workflow_dispatch:
+  push:
+    tags: ['*']
```
Comment (Chrico) [inline .github/workflows/build-to-release-branches.yml:6]: Is not the `master` here missing for updating `master-built` on push?

**Review by Chrico — COMMENTED** [review]
I'm kind of okay with this PR, but there are a few topics which need to be considered and clarify first:

1. Branch protection on "release" and "-built"-branch to avoid direct pushes.
2. Should we always push from `master` to `master-built` on every commit?
3. What about the `1.x`-branch? We should also back-sync it there (in case we merge).
4. Documentation. We need to clarify which branch is what and when to use.

**Comment by luislard** [comment]
> I'm kind of okay with this PR, but there are a few topics which need to be considered and clarify first:
> 
> 1. Branch protection on "release" and "-built"-branch to avoid direct pushes.
> 2. Should we always push from `master` to `master-built` on every commit?
> 3. What about the `1.x`-branch? We should also back-sync it there (in case we merge).
> 4. Documentation. We need to clarify which branch is what and when to use.

:thinking: if we want to propagate this change to 1.x branch I will suggest we only use built branches and trigger the build on branches instead of tags. It will be much simpler. I will make another update to reflect this.

File: .github/workflows/build-to-release-branches.yml (hunk 1 above)
Comment (luislard) [reply .github/workflows/build-to-release-branches.yml]: I added another comment and updated the PR, waiting for your review @Chrico

**Review by Chrico — APPROVED** [review]

**Pull request merged**